server/core/connector.go

```diff
@@ -302,6 +312,64 @@ func (conn *BridgeConnector) subscribeToChannel() (stan.Subscription, error) {
 	return sub, err
 }
 
+// set up a JetStream subscription, assumes the lock is held
+func (conn *BridgeConnector) subscribeToJetStream(subject string) (*nats.Subscription, error) {
+	if conn.bridge.JetStream() == nil {
+		return nil, fmt.Errorf("bridge not configured to use JetStream")
+	}
+
+	options := []nats.SubOpt{}
+
+	if conn.config.DurableName != "" {
+		options = append(options, nats.Durable(conn.config.DurableName))
+	}
+
+	if conn.config.StartAtTime != 0 {
+		t := time.Unix(conn.config.StartAtTime, 0)
+		options = append(options, nats.StartTime(t))
+	} else if conn.config.StartAtSequence == -1 {
+		options = append(options, nats.DeliverLast())
+	} else if conn.config.StartAtSequence > 0 {
+		options = append(options, nats.StartSequence(uint64(conn.config.StartAtSequence)))
+	} else {
+		options = append(options, nats.DeliverAll())
+	}
+
+	options = append(options, nats.AckExplicit())
+	traceEnabled := conn.bridge.Logger().TraceEnabled()
+
+	callback := func(msg *nats.Msg) {
+		start := time.Now()
+		l := int64(len(msg.Data))
+
+		if traceEnabled {
+			conn.bridge.Logger().Tracef("%s received message", conn.String())
+		}
+
+		key := conn.calculateKey(conn.config.Subject, conn.config.DurableName)
+		err := conn.writer(msg).Write(kafka.Message{
+			Key:   key,
+			Value: msg.Data,
+		})
+
+		if err != nil {
+			conn.stats.AddMessageIn(l)
+			conn.bridge.Logger().Errorf("connector publish failure, %s, %s", conn.String(), err.Error())
+		} else {
+			if traceEnabled {
+				conn.bridge.Logger().Tracef("%s wrote message to kafka with key %s", conn.String(), string(key))
+			}
+			msg.Ack()
+			if traceEnabled {
+				conn.bridge.Logger().Tracef("%s acked message to kafka", conn.String())
+			}
+			conn.stats.AddRequest(l, l, time.Since(start))
+		}
+	}
+
+	return conn.bridge.JetStream().Subscribe(subject, callback, options...)
```

Comment:
this means that it would only support push based consumers, the client would error here in case it tries to bind to a pull based consumer. Maybe only push consumers support is fine for now?

Reply:
Yeah, we can definitely add if requested in the future.